src/main/java/org/apache/datasketches/kll/KllItemsSketch.java

```diff
@@ -0,0 +1,366 @@
+    return kllItemsSV.getQuantile(rank, searchCrit);
+  }
+
+  @Override
+  public T[] getQuantiles(final double[] ranks, final QuantileSearchCriteria searchCrit) {
+    if (isEmpty()) { throw new SketchesArgumentException(EMPTY_MSG); }
+    refreshSortedView();
+    final int len = ranks.length;
+    final T[] quantiles = (T[]) Array.newInstance(getMinItem().getClass(), len);
+    for (int i = 0; i < len; i++) {
+      quantiles[i] = kllItemsSV.getQuantile(ranks[i], searchCrit);
+    }
+    return quantiles;
+  }
+
+  @Override
+  public T getQuantileLowerBound(final double rank) {
+    return getQuantile(max(0, rank - KllHelper.getNormalizedRankError(getMinK(), false)));
+  }
+
+  @Override
+  public T getQuantileUpperBound(final double rank) {
+    return getQuantile(min(1.0, rank + KllHelper.getNormalizedRankError(getMinK(), false)));
+  }
+
+  @Override
+  public double getRank(final T quantile, final QuantileSearchCriteria searchCrit) {
+    if (isEmpty()) { throw new SketchesArgumentException(EMPTY_MSG); }
+    refreshSortedView();
+    return kllItemsSV.getRank(quantile, searchCrit);
+  }
+
+  /**
+   * {@inheritDoc}
+   * The approximate probability that the true rank is within the confidence interval
+   * specified by the upper and lower rank bounds for this sketch is 0.99.
+   */
+  @Override
+  public double getRankLowerBound(final double rank) {
+    return max(0.0, rank - KllHelper.getNormalizedRankError(getMinK(), false));
+  }
+
+  /**
+   * {@inheritDoc}
+   * The approximate probability that the true rank is within the confidence interval
+   * specified by the upper and lower rank bounds for this sketch is 0.99.
+   */
+  @Override
+  public double getRankUpperBound(final double rank) {
+    return min(1.0, rank + KllHelper.getNormalizedRankError(getMinK(), false));
+  }
+
+  @Override
+  public double[] getRanks(final T[] quantiles, final QuantileSearchCriteria searchCrit) {
+    if (isEmpty()) { throw new SketchesArgumentException(EMPTY_MSG); }
+    refreshSortedView();
+    final int len = quantiles.length;
+    final double[] ranks = new double[len];
+    for (int i = 0; i < len; i++) {
+      ranks[i] = kllItemsSV.getRank(quantiles[i], searchCrit);
+    }
+    return ranks;
+  }
+
+  @Override
+  public KllItemsSketchSortedView<T> getSortedView() {
+    refreshSortedView();
+    return kllItemsSV;
+  }
+
+  @Override
+  public QuantilesGenericSketchIterator<T> iterator() {
+    return new KllItemsSketchIterator<T>(
+        getTotalItemsArray(), getLevelsArray(SketchStructure.UPDATABLE), getNumLevels());
+  }
+
+  @Override
+  public final void merge(final KllSketch other) {
+    if (readOnly || sketchStructure != UPDATABLE) { throw new SketchesArgumentException(TGT_IS_READ_ONLY_MSG); }
+    final KllItemsSketch<T> othItmSk = (KllItemsSketch<T>)other;
+    if (othItmSk.isEmpty()) { return; }
+    KllItemsHelper.mergeItemImpl(this, othItmSk, comparator);
+    kllItemsSV = null;
+  }
+
+  @Override
+  public void reset() {
```

Comment:
perhaps missing reset of sorted view

Reply:
Thank you for catching this!  
It is now fixed.  see the push notes.